deployment/src/main/java/io/quarkiverse/jackson/jq/deployment/JacksonJqSupport.java

```diff
@@ -0,0 +1,127 @@
+import net.thisptr.jackson.jq.Function;
+import net.thisptr.jackson.jq.VersionRange;
+import net.thisptr.jackson.jq.internal.misc.VersionRangeDeserializer;
+
+final class JacksonJqSupport {
+    private JacksonJqSupport() {
+    }
+
+    static List<JqFuncDef> lookupFunctionsFromConfig(
+            ApplicationArchivesBuildItem archives,
+            JacksonJqConfig config) throws IOException {
+
+        List<JqFuncDef> answer = new ArrayList<>();
+        ObjectMapper mapper = new ObjectMapper();
+
+        for (ApplicationArchive archive : archives.getAllApplicationArchives()) {
+            for (Path root : archive.getRootDirectories()) {
+                final Path resourcePath = root.resolve("net/thisptr/jackson/jq/jq.json");
+
+                if (!Files.exists(resourcePath)) {
+                    continue;
+                }
+                if (!Files.isRegularFile(resourcePath)) {
+                    continue;
+                }
+
+                String functions = Files.readAllLines(resourcePath, StandardCharsets.UTF_8)
+                        .stream()
+                        .filter(l -> !l.isEmpty())
+                        .filter(l -> !l.startsWith("#"))
+                        .collect(Collectors.joining());
+
+                mapper.readValue(functions, JqJson.class).functions.stream()
+                        .filter(f -> f.version.contains(config.functions.version))
+                        .filter(answer::add);
+            }
+        }
+
+        return answer;
+
+    }
+
+    static List<JacksonJqFunctionBuildItem> lookupFunctions(
+            IndexView indexView,
+            JacksonJqConfig config,
+            RecorderContext context,
+            Class<?> annotationType) {
+
+        List<JacksonJqFunctionBuildItem> answer = new ArrayList<>();
+
+        DotName f = DotName.createSimple(Function.class.getName());
+        DotName a = DotName.createSimple(annotationType.getName());
+
+        for (ClassInfo ci : indexView.getAllKnownImplementors(f)) {
+            AnnotationInstance annotation = ci.classAnnotation(a);
+            if (annotation == null) {
+                continue;
+            }
+
+            AnnotationValue values = annotation.value("value");
+            AnnotationValue version = annotation.value("version");
+
+            for (String name : values.asStringArray()) {
+                var function = JacksonJqFunctionBuildItem.of(
+                        name,
+                        context.newInstance(ci.name().toString()));
+
+                if (version != null && !version.asString().isEmpty()) {
+                    if (!VersionRange.valueOf(version.asString()).contains(config.functions.version)) {
+                        continue;
+                    }
+                }
+
+                answer.add(function);
+            }
+        }
+
+        return answer;
+    }
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    public static class JqJson {
```

Comment:
this is the only class that I've somehow copied from jackson-jq at this point, @ricardozanini do you think we can ask the jackson-jq author to make it public ? in such case it would be the only thing we need to be exposed at this point.

Reply:
If we have a good argument with them, it won't be a problem. They are very restricted about leaking their implementation, though.

Reply:
https://github.com/eiiches/jackson-jq/issues/195